### PR TITLE
Safety kit update

### PR DESCRIPTION
**Pull request by omerakkus**

### CODE REVIEW

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyMapper.kt [hunk 1]
```diff
@@ -19,6 +19,6 @@ import com.hms.lib.commonmobileservices.safety.common.Mapper
 
 class GoogleSafetyMapper: Mapper<SafetyNetApi.RecaptchaTokenResponse, SafetyServiceResponse>() {
     override fun map(from: SafetyNetApi.RecaptchaTokenResponse): SafetyServiceResponse = SafetyServiceResponse(
-        responseToken = from.tokenResult
+        responseToken = from.tokenResult!!
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyMapper.kt]: Could you please check that the return type is non-nullable but the method returns an unchecked value?

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt [hunk 2]
```diff
@@ -42,7 +44,7 @@ class GoogleSafetyServiceImpl(private val context: Context): SafetyService {
         SafetyNet.getClient(context).verifyWithRecaptcha(appKey)
             .addOnSuccessListener(){
                 val responseToken = it.tokenResult
-                if(responseToken.isNotEmpty()){
+                if(responseToken!!.isNotEmpty()){
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt]: I think it is better to add null check to the response token.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt [hunk 3]
```diff
@@ -69,13 +71,70 @@ class GoogleSafetyServiceImpl(private val context: Context): SafetyService {
         SafetyNet.getClient(context).attest(nonce, appKey)
             .addOnSuccessListener{ result ->
                 val jwsStr = result.jwsResult
-                val jwsSplit = jwsStr.split(".").toTypedArray()
-                val jwsPayloadStr = jwsSplit[1]
+                val jwsSplit = jwsStr?.split(".")?.toTypedArray()
+                val jwsPayloadStr = jwsSplit!![1]
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt]: I think it is better to add null check.

**Review by oguzozsoy — CHANGES_REQUESTED** [review]
Please check single-line comments.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyMapper.kt (hunk 1 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyMapper.kt]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt (hunk 2 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt (hunk 3 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/google/GoogleSafetyServiceImpl.kt]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt [hunk 4]
```diff
@@ -13,28 +13,52 @@
...
-    fun rootDetection(appKey: String,callback: SafetyRootDetectionCallback<RootDetectionResponse> )
+    fun rootDetection(appKey: String,callback: SafetyRootDetectionCallback<RootDetectionResponse>)
+    fun getMaliciousAppsList(callback: SafetyAppChecksCallback<CommonMaliciousAppResponse>)
+    fun isAppChecksEnabled(callback: SafetyVerifyAppsEnabled<CommonVerifyAppChecksEnabledRes>)
+    fun enableAppsCheck(callback: SafetyVerifyAppsEnabled<CommonVerifyAppChecksEnabledRes>)
+    fun initURLCheck():Work<Unit>
+    fun urlCheck(url:String,appKey: String,threatType:Int,callback:SafetyUrlCheck<CommonUrlCheckRes>)
+    fun shutDownUrlCheck(): Work<Unit>
 
-    interface SafetyServiceCallback<T>{
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt:28]: We have a generic ResultCallback interface in the core module. We can use it instead.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt [hunk 5]
```diff
@@ -13,28 +13,52 @@
...
-        fun onSuccessUserDetect(result: T? = null)
+    interface SafetyServiceCallback<SafetyServiceResponse>{
+        fun onSuccessUserDetect(result: SafetyServiceResponse? = null)
         fun onFailUserDetect(e: java.lang.Exception)
     }
 
-    interface SafetyRootDetectionCallback<T>{
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt:33]: We have a generic ResultCallback interface in the core module. We can use it instead.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt [hunk 6]
```diff
@@ -13,28 +13,52 @@
...
-        fun onSuccessRootDetect(result: T? = null)
+    interface SafetyRootDetectionCallback<RootDetectionResponse>{
+        fun onSuccessRootDetect(result: RootDetectionResponse? = null)
         fun onFailRootDetect(e: java.lang.Exception)
     }
 
+    interface SafetyAppChecksCallback<CommonMaliciousAppResponse>{
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt]: We have a generic ResultCallback interface in the core module. We can use it instead.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt [hunk 7]
```diff
@@ -13,28 +13,52 @@
...
+        fun onSuccessAppChecks(maliciousAppResponse: CommonMaliciousAppResponse? = null)
+        fun onFailAppChecks(e: java.lang.Exception)
+    }
+
+    interface SafetyVerifyAppsEnabled<CommonVerifyAppChecksEnabledRes>{
```
Comment (oguzozsoy) [inline safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt]: We have a generic ResultCallback interface in the core module. We can use it instead.

**Review by oguzozsoy — COMMENTED** [review]
It looks good. But I added some comments to use generic interface in core module.

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt (hunk 4 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt:28]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt (hunk 5 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt:33]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt (hunk 6 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt]: resolved

File: safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt (hunk 7 above)
Comment (omerakkus) [reply safety/src/main/java/com/hms/lib/commonmobileservices/safety/SafetyService.kt]: resolved

**Review by oguzozsoy — APPROVED** [review]
All looks good

**Pull request merged**